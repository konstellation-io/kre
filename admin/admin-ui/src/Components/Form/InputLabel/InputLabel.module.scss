@import 'Styles/mixins';
@import 'Styles/colors';

.label {
  @include font-caption;

  display: block;
  letter-spacing: 0.4px;
  color: font-color(grey);
  margin-bottom: 11px;
  min-height: 13px;

  transition: color ease 0.3s;

  &.hidden {
    color: transparent;
  }
}
