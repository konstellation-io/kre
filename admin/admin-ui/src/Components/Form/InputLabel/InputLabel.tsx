import React from 'react';
import cx from 'classnames';
import styles from './InputLabel.module.scss';

type Props = {
  text?: string;
  hidden?: boolean;
};

function InputLabel({ text = '', hidden = false }: Props) {
  return (
    <label
      className={cx(styles.label, { [styles.hidden]: hidden })}
      data-testid="label"
    >
      {text.toUpperCase()}
    </label>
  );
}

export default InputLabel;
