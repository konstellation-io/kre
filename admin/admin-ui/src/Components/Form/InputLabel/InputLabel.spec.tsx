import React from 'react';
import { shallow } from 'enzyme';
import InputLabel from './InputLabel';

describe('InputLabel', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<InputLabel text="some message" />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
