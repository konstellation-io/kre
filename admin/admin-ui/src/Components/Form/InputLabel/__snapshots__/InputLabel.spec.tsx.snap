// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel matches snapshot 1`] = `
<label
  className="label"
  data-testid="label"
>
  SOME MESSAGE
</label>
`;
